### DISCUSSION TITLE
请问一下关于nginx反向代理的问题

### DISCUSSION THREAD

**Discussion by l987845225** [AlistGo/alist#3758] [Q&A]
## 通过nginx转发之后，直接用80访问就显示这样<font style=background:red>↓↓↓</font>，打不开主页
```
System error: TypeError: Failed to fetch dynamically imported module: http://xxxxxx.eu.org/alist/assets/index.7c4a1f9d.js
System error: TypeError: Failed to fetch dynamically imported module: http://192.168.68.154/alist/assets/index.7c4a1f9d.js
```
![](https://article.biliimg.com/bfs/article/44c669d4f23ef1ef646f18f9d2ca6301a450dc34.png)
## nginx.conf也改了：
```js
worker_processes  1;
events {
    worker_connections  1024;
}
http {
    include       mime.types;
    default_type  application/octet-stream;
    sendfile        on;
    keepalive_timeout  65;

    server {
        listen       80;
        server_name  localhost;
 
	 location / {
	  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	  proxy_set_header X-Forwarded-Proto $scheme;
	  proxy_set_header Host $http_host;
	  proxy_set_header X-Real-IP $remote_addr;
	  proxy_set_header Range $http_range;
	  proxy_set_header If-Range $http_if_range;
	  proxy_redirect off;
	  proxy_pass http://192.168.68.154:5244;#用的docker127.0.0.1不能用
	  client_max_body_size 20000m;
	}
```

**Comment by l987845225** [comment]
回下自己
nginx个人无解了，转头用Apache一次性解决，
Apache碰到两个问题：
1. 因为用的docker没有在挂载目录生成配置文件，去找了其他人分享的，但是用不了，最后还是在镜像里找到了文件
2. 官方给的配置文件有“/” ‘“http://xxxxx.xxx.xxx”’ ,引号要去到，不然容器秒退

**Comment by mengguyi** [comment]
https://github.com/alist-org/alist/commit/6c2f3486fc155e795026d08cf55d559f3e63595a
或许有帮助

**Comment by KafCoppelia** [comment]
同样根据文档配置nginx的子目录反代，通过 `domain/alist` 访问不了

**Comment by mouccc168** [comment]
最近成功配置子目录
 location ^~ /alist/ {
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header Range $http_range;
      proxy_set_header If-Range $http_if_range;
      proxy_redirect off;
      proxy_pass http://127.0.0.1:5244/alist/;
      # the max size of file to upload
      client_max_body_size 20000m;
}
多加^~ 强制匹配，优先级高

**Comment by Troray** [comment]
#### 根据你给出的信息来看
- 你是docker部署的
- 你想反代到二级目录 `http://www.yourdomain.com/alist`

#### `etc/alist/config.json`需要修改的地方：
- `port` :可以保持`5244`默认不修改，除非该端口已被占用。(新的版本是`http_port`)
- `"site_url": "/alist"` :根据你实际情况自行设置

#### /etc/nginx/nginx.conf
1. 查看`nginx.conf` 中是否有`include /etc/nginx/modules-enabled/*.conf;` 或者 `include /etc/nginx/conf.d/*.conf;`  
2. 去对应的`modules-enabled`或者`conf.d`目录下新建一个配置文件，例如`alist.conf`
3. 将以下内容copy到你刚创建的`alist.conf`文件中
```
location /alist/ {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Range $http_range;
	  proxy_set_header If-Range $http_if_range;
    proxy_redirect off;
    proxy_pass http://127.0.0.1:5244/alist/;
    # the max size of file to upload
    client_max_body_size 20000m;
}
```
4. `systemctl reload nginx`后应该就能生效正常使用了。
5. 配置截图：
![配置](https://i.imgur.com/a5C6da2.png)

7. 最终效果：
![alist](https://i.imgur.com/FcJoie9.png)

**Reply by JerryChenn07** [reply]
再强调下，基于这个nginx配置这里site_url要修改

/etc/alist/config.json需要修改的地方：
port :可以保持5244默认不修改，除非该端口已被占用。(新的版本是http_port)
"site_url": "/alist" :根据你实际情况自行设置

**Comment by ArTreCry** [comment]
不想用子目录也可以二级域名
```shell
server {
listen 80;  #监听80
server_name xxx.xxx.com;  #二级域名访问
location / {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;    
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Range $http_range;
    proxy_set_header If-Range $http_if_range;
    proxy_redirect http://xxx.xxx.com:5244/ /;   #反代二级域名
    proxy_pass http://127.0.0.1:5244/;
    # the max size of file to upload
    client_max_body_size 20000m;
}
}
```
记得解析
